cli/tests/services.bats

```diff
@@ -554,3 +570,213 @@ EOF
+  export FLOX_FEATURES_SERVICES=true
+
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [services]
+    one.command = "sleep infinity"
+EOF
+  )"
+
+  "$FLOX_BIN" init
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f -
+
+  SCRIPT="$(cat << "EOF"
+    # don't set -euo pipefail because we expect these to fail
+    "$FLOX_BIN" services start one invalid
+    "$FLOX_BIN" services status
+EOF
+  )"
+
+  run "$FLOX_BIN" activate -- bash -c "$SCRIPT"
+  assert_failure
+  assert_output --partial "Service 'invalid' not found."
+  assert_output --partial "couldn't connect to service manager"
+}
+
+# Also tests service names with spaces in them, because starting them is handled
+# in Bash
+@test "start: only starts specified services" {
+
+  export FLOX_FEATURES_SERVICES=true
+
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [services]
+    no_space.command = "sleep infinity"
+    "with space".command = "sleep infinity"
+    skip.command = "sleep infinity"
+EOF
+  )"
+
+  "$FLOX_BIN" init
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f -
+
+  SCRIPT="$(cat << "EOF"
+    set -euo pipefail
+
+    "$FLOX_BIN" services start no_space "with space"
+    "$FLOX_BIN" services status
+EOF
+  )"
+
+  run "$FLOX_BIN" activate -- bash -c "$SCRIPT"
+  assert_success
+  assert_output --partial "Service 'no_space' started."
+  assert_output --partial "Service 'with space' started."
+  assert_output --partial "no_space   Running"
+  assert_output --partial "with space Running"
+  assert_output --partial "skip       Disabled"
+}
+
+@test "start: defaults to all services" {
+
+  export FLOX_FEATURES_SERVICES=true
+
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+
+    [services]
+    one.command = "sleep infinity"
+    two.command = "sleep infinity"
+EOF
+  )"
+
+  "$FLOX_BIN" init
+  echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -f -
+
+  SCRIPT="$(cat << "EOF"
+    set -euo pipefail
+
+    "$FLOX_BIN" services start
+    "$FLOX_BIN" services status
+EOF
+  )"
+
+  run "$FLOX_BIN" activate -- bash -c "$SCRIPT"
+  assert_success
+  assert_output --partial "Service 'one' started."
+  assert_output --partial "Service 'two' started."
+  assert_output --partial "one        Running"
+  assert_output --partial "two        Running"
+}
+
+@test "start: picks up changes after environment modification when all services have stopped" {
+
+  export FLOX_FEATURES_SERVICES=true
+
+  MANIFEST_CONTENTS_1="$(cat << "EOF"
+    version = 1
+
+    [services]
+    one.command = "echo $FOO"
+
+    [hook]
+    on-activate = "export FOO=foo_one"
+EOF
+  )"
+
+  "$FLOX_BIN" init
+  echo "$MANIFEST_CONTENTS_1" | "$FLOX_BIN" edit -f -
+
+  # Edit the manifest adding a second service and changing the value of FOO.
+  # Then start services again.
+  run "$FLOX_BIN" activate -s -- bash "${TESTS_DIR}/services/start_picks_up_modifications.sh"
+  assert_success
+
+  # The added service should be running.
+  assert_output --partial "two        Running"
```

Comment:
Just FYI you can use a regex with the `--regex` flag and not need to get the spacing exactly right.

Reply:
Happy to change if you feel strongly - I feel like brittle spaces are pretty easy to fix